### PR TITLE
add truffleruby dockerfiles

### PR DESCRIPTION
**Pull request by abdelhaira**
Add Truffleruby images + Readme

### CODE REVIEW

**Review by eregon — APPROVED** [review]
Looks good, thanks!

File: truffleruby/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# LICENSE UPL 1.0
+#
+# Copyright (c) 2021 Oracle and/or its affiliates. All rights reserved.
```
Comment (eregon) [inline truffleruby/Dockerfile:3]: AFAIK it's no longer necessary to write `All rights reserved.`, and that also tends to be confusing (cc @wirthi).
Could you drop it here? (and in other files in this PR)

Comment (wirthi) [reply truffleruby/Dockerfile:3]: This is the case I understand. Internal link: http://pdb.us.oracle.com/downloads/legal_notices.html

**Comment by ezzarghili** [comment]
Please integrate the small changes in https://github.com/graalvm/container/pull/12 and  https://github.com/graalvm/container/pull/13, no need to have separate PRs for those.

File: truffleruby/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    && curl --fail --silent --location --retry 3 https://github.com/oracle/truffleruby/releases/download/vm-$GRAALVM_VERSION/truffleruby-$GRAALVM_VERSION-linux-amd64.tar.gz \
+    | gunzip | tar x -C /opt/
+
+ENV LANG=en_US.UTF-8 \
+    PATH=/opt/truffleruby-$GRAALVM_VERSION-linux-amd64/bin:$PATH
+
+RUN echo "gem: --no-document" > ~/.gemrc
+RUN /opt/truffleruby-$GRAALVM_VERSION-linux-amd64/lib/truffle/post_install_hook.sh
+
+CMD ruby --version
```
Comment (ezzarghili) [inline truffleruby/Dockerfile:31]: Let's `ruby` as ENTRYPOINT, `--version` as CMD

File: truffleruby/Dockerfile.ol8 [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    && curl --fail --silent --location --retry 3 https://github.com/oracle/truffleruby/releases/download/vm-$GRAALVM_VERSION/truffleruby-$GRAALVM_VERSION-linux-amd64.tar.gz \
+    | gunzip | tar x -C /opt/
+
+ENV LANG=en_US.UTF-8 \
+    PATH=/opt/truffleruby-$GRAALVM_VERSION-linux-amd64/bin:$PATH
+
+RUN echo "gem: --no-document" > ~/.gemrc
+RUN /opt/truffleruby-$GRAALVM_VERSION-linux-amd64/lib/truffle/post_install_hook.sh
+
+CMD ruby --version
```
Comment (ezzarghili) [inline truffleruby/Dockerfile.ol8:31]: same here, Let's  use `ruby` as ENTRYPOINT, `--version` as CMD

File: truffleruby/Dockerfile.ol8-slim [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+RUN set -eux \
+    && curl --fail --silent --location --retry 3 https://github.com/oracle/truffleruby/releases/download/vm-$GRAALVM_VERSION/truffleruby-$GRAALVM_VERSION-linux-amd64.tar.gz \
+    | gunzip | tar x -C /opt/ \
+    && rm -rf /opt/truffleruby-$GRAALVM_VERSION-linux-amd64/lib/llvm-toolchain
+
+ENV LANG=en_US.UTF-8 \
+    PATH=/opt/truffleruby-$GRAALVM_VERSION-linux-amd64/bin:$PATH
+
+
+CMD ruby --version
```
Comment (ezzarghili) [inline truffleruby/Dockerfile.ol8-slim:30]: same here, Let's use `ruby` as ENTRYPOINT, `--version` as CMD

File: truffleruby/Dockerfile.slim [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+RUN set -eux \
+    && curl --fail --silent --location --retry 3 https://github.com/oracle/truffleruby/releases/download/vm-$GRAALVM_VERSION/truffleruby-$GRAALVM_VERSION-linux-amd64.tar.gz \
+    | gunzip | tar x -C /opt/ \
+    && rm -rf /opt/truffleruby-$GRAALVM_VERSION-linux-amd64/lib/llvm-toolchain
+
+ENV LANG=en_US.UTF-8 \
+    PATH=/opt/truffleruby-$GRAALVM_VERSION-linux-amd64/bin:$PATH
+
+CMD ruby --version
```
Comment (ezzarghili) [inline truffleruby/Dockerfile.slim:29]: same here, Let's use `ruby` as ENTRYPOINT, `--version` as CMD

**Comment by abdelhaira** [comment]
Done

**Review by ezzarghili — CHANGES_REQUESTED** [review]

**Review by ezzarghili — APPROVED** [review]

**Pull request merged**